operators/src/processing/temporal_raster_aggregation/mean_aggregation_subquery.rs

```diff
@@ -125,13 +126,14 @@ impl<T> TemporalMeanTileAccu<T> {
     }
 }
 
+#[async_trait]
 impl<T> FoldTileAccu for TemporalMeanTileAccu<T>
 where
     T: Pixel,
 {
     type RasterType = T;
 
-    fn into_tile(self) -> RasterTile2D<Self::RasterType> {
+    async fn into_tile(self) -> Result<RasterTile2D<Self::RasterType>> {
```

Comment:
not necessarily. I do spawn a thread in the interpolation operator but it could as well perform some I/O operation or something else that can be `awaited`. I guess we could simplify some code in the subquery adapter by using async_trait instead of methods that produce Futures too.